events/api.py

```diff
@@ -3618,11 +3627,39 @@ class EventFilter(django_filters.rest_framework.FilterSet):
         widget=django_filters.widgets.CSVWidget(),
         method=partial(in_or_null_filter, "super_event"),
     )
+    dwithin = django_filters.Filter(
+        method="filter_dwithin", widget=DistanceWithinWidget()
+    )
 
     class Meta:
         model = Event
         fields = ("division", "super_event_type", "super_event")
 
+    @staticmethod
+    def filter_dwithin(queryset, name, value: tuple[tuple[str, str], str]):
+        origin, metres = value
+        if not (origin and metres):
+            # Need both values for filtering
+            return queryset
+
+        try:
+            origin_x, origin_y = [float(i) for i in origin]
+        except ValueError:
+            raise ParseError("Origin must be a tuple of two numbers")
+        try:
+            metres = float(metres)
+        except ValueError:
+            raise ParseError("Metres must be a number")
+
+        places = Place.geo_objects.filter(
```

Comment:
I think `geo_objects` is just some relic from the past, and you could and should use just `objects` instead.